packages/docusaurus-plugin-openapi/src/sidebars/index.ts

```diff
@@ -59,6 +68,9 @@ export function generateSidebars(items: Item[], options: Options): PropSidebar {
       const info = prototype?.api?.info;
       const fileName = path.basename(source).split(".")[0];
       return {
+        source: prototype?.source,
+        sourceDirName: prototype?.sourceDirName ?? ".",
```

Comment:
This makes me a little uncomfortable having the existence of source being tied to the existence of `api.info`. Maybe we should just use `items[0].source` and `items[0].sourceDirName`? However, with hardcoding zero index we have to think if there is ever a case were `items` is an empty array? Maybe an empty directory, or spec that wouldn't generate any pages, etc... We should probably have a case that skips creating the category/subitems if we don't have the right info

Reply:
> If there is ever a case were items is an empty array?

I don't think so but can mull on it. 

The current implementation (for all of multi-spec) relies on the folder structure being generated from OpenAPI Specs (& [these lines are only hit if there are > 1 spec files)](https://github.com/blugavere/docusaurus-plugin-openapi/blob/8161387ad207f38dcb80828d919b24f7f7a9a941/packages/docusaurus-plugin-openapi/src/sidebars/index.ts#L85). 

Arbitrary markdown files aren't supported. Introductions are automatically generated from the `info.description` field so in all cases a section is only created for an openapi spec. [Here](https://github.com/blugavere/docusaurus-plugin-openapi/blob/8161387ad207f38dcb80828d919b24f7f7a9a941/packages/docusaurus-plugin-openapi/src/sidebars/index.ts#L66) we're ignoring the auto-generated markdown file from being used to select the sidebar label after being grouped by [source](https://github.com/blugavere/docusaurus-plugin-openapi/blob/8161387ad207f38dcb80828d919b24f7f7a9a941/packages/docusaurus-plugin-openapi/src/sidebars/index.ts#L63).

The fn input items[] is a flattened array of ["auto-generated markdown for spec a", "openapi path for spec a"]. [The .find fn ](https://github.com/blugavere/docusaurus-plugin-openapi/blob/8161387ad207f38dcb80828d919b24f7f7a9a941/packages/docusaurus-plugin-openapi/src/sidebars/index.ts#L65) is getting not the zero index, but the first element in the group which is non-markdown & has the openapi spec info. [If falsy, defaults to filename](https://github.com/blugavere/docusaurus-plugin-openapi/blob/8161387ad207f38dcb80828d919b24f7f7a9a941/packages/docusaurus-plugin-openapi/src/sidebars/index.ts#L77).